src/main/java/microsoft/exchange/webservices/data/ExchangeServiceBase.java

```diff
@@ -554,21 +507,30 @@ protected Date convertUniversalDateTimeStringToDate(String dateString) {
   }
 
   /**
-   * Converts xs:dateTime string with either "Z", "-00:00" bias, or ""
-   * suffixes to unspecified StartDate value ignoring the suffix.Needs to fix
-   * E14:232996.
-   *
-   * @param value The string value to parse.
-   * @return The parsed DateTime value.
+   * Tries to parse the dateString using the given patterns. If no timezone is
+   * defined in the dateString, you can choose to default to UTC. If none of 
+   * the patterns can parse the dateString, and Exception is thrown.
+   * 
+   * @param dateString dateString to parse
+   * @param isUTC if no timezone if defined in dateString, default to UTC
+   * @param patterns patterns to try for parsing. 
+   * @return parsed Date. 
+   * @throws IllegalArgumentException if no patterns can parse the dateString
    */
-  protected Date convertStartDateToUnspecifiedDateTime(String value)
-      throws ParseException {
-    if (value == null || value.isEmpty()) {
-      return null;
-    } else {
-      DateFormat df = new SimpleDateFormat("yyyy-MM-dd'Z'");
-      return df.parse(value);
+  Date parseDateTimeString(String dateString, boolean isUTC, String... patterns) {
+    ParseException lastException = null;
```

Comment:
I have now added aggregated results to explain more which patterns were tried when parsing.